index.js

```diff
@@ -1,21 +1,593 @@
+
+
+    const itrEnabled = true;
+    if (itrEnabled) {
+      //If project uses testplan force use of code coverage
+      let file_list = recFindByExt(".", "xctestplan");
+      for (let testPlanFile of file_list) {
+        await deleteLinesContaining(testPlanFile, "codeCoverage");
+      }
+    }
+
+    //Create folder to store files
+    if (!fs.existsSync(sdkTestingDir)) {
+      fs.mkdirSync(sdkTestingDir);
+    }
+
+    //Read project
+    const workspace = await getWorkspace();
+    let xcodeproj = await getXCodeProj();
+    var projectParameter;
+
+    //download testing framework
+    await downloadLatestFramework(libraryVersion);
 
-    core.debug((new Date()).toTimeString()); // debug is only output if you set the secret `ACTIONS_RUNNER_DEBUG` to true
-    await wait(parseInt(ms));
-    core.info((new Date()).toTimeString());
+    if (workspace) {
+      console.log(`Workspace selected: ${workspace}`);
+      projectParameter = "-workspace " + `"${workspace}"`;
+    } else if (xcodeproj) {
+      console.log(`Project selected: ${xcodeproj}`);
+      projectParameter = "-project " + `"${xcodeproj}"`;
+    } else if (fs.existsSync("Package.swift")) {
+      console.log(`Package.swift selected`);
+      await swiftPackageRun(platform, extraParameters, itrEnabled);
+      return;
+    } else {
+      core.setFailed(
+        "Unable to find workspace, project or Swift package file. Please set with workspace or xcodeproj"
+      );
+    }
+
+    const scheme = await getScheme(workspace, xcodeproj);
+    console.log(`Scheme selected: ${scheme}`);
+
+      //copy configfile
+      const configfileName = "ddTesting.xcconfig";
+      const configFilePath = sdkTestingDir + "/" + configfileName;
+      createXCConfigFile(configFilePath, sdkTestingDir);
+
+
+
+    let codeCoverParam = "";
+    if (itrEnabled) {
+      codeCoverParam = "-enableCodeCoverage YES";
+    }
+
+    //build for testing
+    let buildCommand =
+      "xcodebuild build-for-testing " +
+      codeCoverParam +
+      " -xcconfig " +
+      configFilePath +
+      " " +
+      projectParameter +
+      " -configuration " +
+      configuration +
+      " -scheme " +
+      `"${scheme}"` +
+      " -sdk " +
+      sdk +
+      " -derivedDataPath " +
+      derivedDataPath +
+      " -destination " +
+      `"${destination}" ` +
+      extraParameters;
+    const result = await exec.exec(buildCommand, null, null);
+
+    //For all testruns that are configured
+    let testRuns = await getXCTestRuns();
+    let testError;
+
+    for (const testRun of testRuns) {
+      //modify xctestrun with Scope variables
+
+      let plutilExportCommand =
+        "plutil -convert json -o " + testrunJson + ` "${testRun}"`;
+      await exec.exec(plutilExportCommand, null, null);
+
+      let jsonString = fs.readFileSync(testrunJson, "utf8");
+      const testTargets = JSON.parse(jsonString);
+
+      for (const target of Object.keys(testTargets)) {
+        if (target.charAt(0) !== "_") {
```

Comment:
what does it mean that the first character of `target` is `'_'`?

Reply:
The keys I am looking for are always identified like this, I can skip the others

Reply:
does that have a reason? I'm looking for a way to make this easier to read in the future. Is it because you ignore the private properties? 

Also, this would be clearer:
```javascript
const testTargetsToModify = Object.keys(testTargets).filter(key => key[0] !== '_')
```
and then you can remove the `if (target.charAt(0) !== "_") {` at the beginning of the for loop

Reply:
Applied your solution